apps/frontend/src/components/global/Sidebar.vue

```diff
@@ -90,36 +157,115 @@ const SidebarProps = Vue.extend({
   components: {
     LinkItem,
     FileItem,
+    ProfileItem,
     AboutModal,
     HelpModal
   }
 })
 export default class Sidebar extends SidebarProps {
-  //selects/deselects all files. Will select all unless all are already selected
-  toggle_all(): void {
-    if (this.all_toggled) {
-      FilteredDataModule.set_toggled_files([]);
+  // used to toggle v-exapansion-panel
+  file_views = 0;
+
+  created() {
+    // open the appropriate v-expansion-panel based on current route
+    if (this.curr_route_path == '/results') this.file_views = 0;
+    else if (this.curr_route_path == '/compare') this.file_views = 0;
+    else if (this.curr_route_path == '/profiles') this.file_views = 1;
+  }
+
+  // select the specified files
+  set_selected_files(files: number[]): void {
+    FilteredDataModule.set_toggled_files([]);
+    FilteredDataModule.set_toggled_files(files);
+  }
+
+  // toggle the "select all" for profiles
+  toggle_all_profiles(): void {
+    if (this.all_toggled_profiles) {
+      this.set_selected_files(this.selected_evaluations);
     } else {
-      FilteredDataModule.set_toggled_files(
-        InspecDataModule.allFiles.map(v => v.unique_id)
-      );
+      let files = InspecDataModule.allProfileFiles.map(v => v.unique_id);
+      files.push(...this.selected_evaluations);
+      this.set_selected_files(files);
```

Comment:
Making this change will affect code such as select_file in SidebarFile.  Instead of sending a file to be toggled on or off, an array of one element will need to be created and sent.

Instead, I changed
```ruby
this.set_selected_files(files);
```
to
```ruby
FilterdDataModule.set_toggled_files(files);
```
where files is an array